### DISCUSSION TITLE
PayGreenJS - Fonctionnement du mode `instrument`

### DISCUSSION THREAD

**Discussion by francoistomasi** [PayGreen/documentation#7] [Q&A] [answered]
Bonjour,

Je suis en train d'intégrer le un système d'enregistrement d'instruments avec la v3 de PayGreen.

En utilisant le mode `instrument` [Voir doc](https://developers.paygreen.fr/docs/paygreenjs-modes#mode-instrument), est-t-il possible d'avoir un fonctionnement ou la carte est réutilisable sans que la case soit à cocher et ainsi la masquer.

Car avec le fonctionnement actuel, ce mode qui a pour but d'enregistrer une carte afin de la réutiliser sans redemander les infos du client, quel est l'objectif de ne pas la rendre obligatoirement réutilisable ?

**Comment by enzobillisPG** [comment]
Bonjour,

La case à cocher permet de recueillir le consentement de l'acheteur quant à l'enregistrement de sa carte et sa réutilisation par le marchand. Le côté réutilisable de l'instrument dépend donc uniquement du consentement. 

Dans certaines intégrations un instrument non réutilisable est suffisant car il n'y a pas de captures prévues.

Bonne journée,
Enzo

**Comment by francoistomasi** [comment]
Bonjour,

Merci pour ces précisions, cependant je ne comprends pas trop l'intérêt du mode `instrument` sans possibilité de réutilisation derrière 🤔

Ce mode n'est-il pas là uniquement pour remplacer l'ancien mode `tokenizer` qui visait à réutiliser une carte ?

J'ai aussi vu dans la documentation qu'il était possible de masquer cette case à cocher via un paramètre `updateConsent`

[Voir doc](https://developers.paygreen.fr/docs/paygreenjs-installation#3-init)
updateConsent | Change the actual state of the consent.Accept a boolean true/false parameter which represents if the user consent.(can be used only if you passed an technical audit by Paygreen)
-- | --

Merci d'avance,

**Comment by enzobillisPG** [comment, marked as answer]
Quand on crée un instrument, si ce dernier n'est pas réutilisable (consentement refusé) alors l'instrument ne sera utilisable qu'une seule fois. Si l'option `authorizedInstrument` est désactivée, cela ne déclenchera pas de 3DS ou OAuth et l'instrument pourra être utilisé pour une autorisation/capture. Si elle est activée, cela déclenchera la sécurité et consommera l'unique utilisation autorisée de l'instrument. Il sera alors "vérifié" mais ne pourra plus être utilisé pour une autorisation/capture plus tard.

Cela offre donc deux types d'intégration pour les instruments non réutilisables:
- Les instruments réutilisables qu'une seule fois, non vérifiés
- Les instruments vérifiés mais plus utilisables

Si vous ne souhaitez pas conserver ce fonctionnement vous pouvez tout à fait, grâce à l'event `REUSABLE_ALLOWED_CHANGE` suivre l'état de la checkbox et disable/enable votre bouton de paiement. Cela rendra obligatoire le consentement pour enregistrer sa carte. 

Enfin, la méthode `updateConsent` est en effet soumise à un audit technique de votre page de paiement afin de s'assurer que le consentement reste bien recueilli. Je vous laisse vous rapprocher du support client pour cela.

Bonne journée,

**Reply by francoistomasi** [reply]
Et quelle est la différence entre un instrument `vérifié` et un `non vérifié` si dans les 2 cas l'instrument n'est utilisable qu'une seule fois ?

**Reply by enzobillisPG** [reply]
Un instrument vérifié va avoir été authentifié auprès de le banque (3DS) ou de l'entreprise pour le cas des moyens de paiement wallets (OAuth). Ce qui consommera la seule et unique utilisation possible si l'instrument n'est pas réutilisable.
